File: code/c/storing-using-data/variable_test.c
/* Program: variable_test.c
*  This program demonstrates some variable declarations.
*/
#include <stdio.h>

const float PI = 3.1415;
float global_float = 12.3;
int global_int = 73;

void test(int param_int, float param_float)
{
    int my_local = 37, another_local = 42;
    printf("my local int = %d, another_local = %d\n", my_local, another_local);
    printf("param int = %d, param float = %f\n", param_int, param_float);
    printf("globals are %f and %d\n", global_float, global_int);
}

int main()
{
    int local_int;
    local_int = 21;
    
    test(local_int, PI * local_int * local_int);
    
    printf("local int = %d\n", local_int);
    printf("globals are %f and %d\n", global_float, global_int);
    printf("PI is a constant with value %f\n", PI);
    return 0;
}

File: code/pascal/storing-using-data/VariableTest.pas
//  This program demonstrates some variable declarations.
program VariableTest;

const PI = 3.1415;
var 
    globalFloat: Single = 12.3;
    globalInt: Integer = 73;

procedure Test(paramInt: Integer; paramFloat: Single);
var
    myLocal: Integer = 37;
    anotherLocal: Integer = 42;
begin
    WriteLn('my local int = ', myLocal, ', anotherLocal = ', anotherLocal);
    WriteLn('param int = ', paramInt, ', param int2 = ', paramFloat:4:2);
    WriteLn('globals are ', globalFloat, ' and ', globalInt);
end;

procedure Main();
var
    localInt: Integer;
begin
    localInt := 21;
    
    Test(localInt, PI * localInt * localInt);
    
    WriteLn('local int = ', localInt);
    WriteLn('globals are ', globalFloat, ' and ', globalInt);
    WriteLn('PI is a constant with value ', PI);
end;

begin
    Main();
end.